packages/tutorial/src/part6.tsx

```diff
@@ -0,0 +1,124 @@
+function AnswerQuestion({ question, animal }: { question: string; animal: string }) {
+  const exactRoute = `The user asks if the animal is ${animal}.`;
+  const correctRoute = `The user's query is true for ${animal}`;
+  const incorrectRoute = `The user's query is false for ${animal}`;
+
+  return (
+    <NaturalLanguageRouter query={question}>
+      <Route when={exactRoute}>
+        <ChatCompletion>
+          <SystemMessage>
+            Rewrite this statement to be grammatically correct, without quotation marks, using the form: "🎉 You got it
+            right! The animal I was thinking of was..."
+          </SystemMessage>
+          <UserMessage>🎉 You got it right! The animal I was thinking of was {animal}.</UserMessage>
+        </ChatCompletion>
+      </Route>
+      <Route when={correctRoute}>
+        <ChatCompletion>
+          <SystemMessage>
+            Answer the following question about the animal "{animal}" in the affirmative, using the form: "👍 Yes, the
+            animal I am thinking of ...". Do not, under any circumstances, reveal any other information, especially not
+            the animal itself.
+          </SystemMessage>
+          <UserMessage>{question}</UserMessage>
+        </ChatCompletion>
+      </Route>
+      <Route when={incorrectRoute}>
+        <ChatCompletion>
+          <SystemMessage>
+            Answer the following question about the animal "{animal}" in the negative, using the form: "❌ No, the
+            animal I am thinking of ...". Do not, under any circumstances, reveal any other information, especially not
+            the animal itself.
+          </SystemMessage>
+          <UserMessage>{question}</UserMessage>
+        </ChatCompletion>
+      </Route>
+    </NaturalLanguageRouter>
+  );
+}
+
+function AnimalGame({ question, animal }: { question: string; animal: string }) {
+  return (
+    <QuestionFilter question={question}>
+      <AnswerQuestion question={question} animal={animal} />
+    </QuestionFilter>
+  );
+}
+
+// The below is the main game logic.
+
+// First, come up with the animal we're going to get the user to guess.
+const renderContext = AI.createRenderContext();
+let animal = await renderContext.render(<ThinkOfAnAnimal />);
+if (animal.endsWith('.')) {
+  animal = animal.slice(0, -1);
+}
+// Of course, this is for debugging only!
+console.log(`I am thinking of: ${animal}\n`);
+
+// Now, let the user ask up to 20 questions about the animal.
+let numQuestions = 0;
+const MAX_QUESTIONS = 20;
+while (numQuestions < MAX_QUESTIONS) {
+  const question = await getQuestion(numQuestions, MAX_QUESTIONS);
+  const answer = await renderContext.render(<AnimalGame question={question} animal={animal} />);
+  console.log(`${answer}\n`);
+  numQuestions++;
+  if (answer.includes('You got it right!')) {
+    console.log('You win!');
+    break;
+  }
+}
+
+if (numQuestions === MAX_QUESTIONS) {
+  console.log(`I'm sorry, the animal I was thinking of was a ${animal}.`);
```

Comment:
minor: We could do this with model call as well to get the `a` vs `an` phrasing correct.